v1/src/test/java/com/google/cloud/teleport/splunk/HttpEventPublisherTest.java

```diff
@@ -116,14 +122,24 @@ public void stringPayloadTest()
 
     String actual = publisher.getStringPayload(SPLUNK_EVENTS);
 
-    String expected =
-        "{\"time\":12345,\"host\":\"test-host-1\",\"source\":\"test-source-1\","
-            + "\"sourcetype\":\"test-source-type-1\",\"index\":\"test-index-1\","
-            + "\"event\":\"test-event-1\"}{\"time\":12345,\"host\":\"test-host-2\","
-            + "\"source\":\"test-source-2\",\"sourcetype\":\"test-source-type-2\","
-            + "\"index\":\"test-index-2\",\"event\":\"test-event-2\"}";
-
-    assertThat(expected, is(equalTo(actual)));
+    List<JsonElement> expectedElements = new ArrayList<>();
+    expectedElements.add(
+        JsonParser.parseString(
+            "{\"time\":12345,\"host\":\"test-host-1\",\"source\":\"test-source-1\","
+                + "\"sourcetype\":\"test-source-type-1\",\"index\":\"test-index-1\","
+                + "\"event\":\"test-event-1\"}"));
+    expectedElements.add(
+        JsonParser.parseString(
+            "{\"time\":12345,\"host\":\"test-host-2\","
+                + "\"source\":\"test-source-2\",\"sourcetype\":\"test-source-type-2\","
+                + "\"index\":\"test-index-2\",\"event\":\"test-event-2\"}"));
+
+    String[] jsonActual = actual.split("(?<=\\})(?=\\{)");
```

Comment:
@zzjas  I don't think so, the ```actual``` is a string which contains 2 json objects -> converted to string -> concatenated. Therefor I am trying to separate them out by manually parsing it. Existing libraries as per my knowledge can work for a single json string and not for multiple json strings concatenated. I believe it gives JSON malformed error. 

Do you have any better approach in mind?

Reply:
One idea is probably to return two JSON strings instead of concatenating them. It might require complicated changes to the main code so manually parsing the output is ok.